lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -753,4 +727,122 @@ sub _create_download_lists {
+    my $data = eval { load_yaml(@load_yaml_args) };
+    if (my $error = $@) { return {error_message => "Unable to load YAML: $error"} }
+    my $app = OpenQA::App->singleton;
+    my $validation_errors = $app->validate_yaml($data, 'JobScenarios-01.yaml', $app->log->level eq 'debug');
+    return {error_message => "YAML validation failed:\n" . join("\n", @$validation_errors)} if @$validation_errors;
+
+    my $products = $data->{products};
+    my $machines = $data->{machines};
+    my $job_templates = $data->{job_templates};
+    my $error_msg;
+    my %wanted;
+    my @job_templates;
+    foreach my $key (keys %$job_templates) {
+        my $job_template = $job_templates->{$key};
+        next unless my $product_name = $job_template->{product};
+        next unless my $product = $products->{$product_name};
+        next
+          if ( $product->{distri} ne _distri_key($args)
+            || $product->{flavor} ne $args->{FLAVOR}
+            || $product->{version} ne $args->{VERSION}
+            || $product->{arch} ne $args->{ARCH});
+        my $settings = $job_template->{settings} // {};
+        $settings->{TEST} = $key;
+        my @worker_class;
+        push @worker_class, $settings->{WORKER_CLASS} if $settings->{WORKER_CLASS};
+
+        my $product_settings = $product->{settings} // {};
+        foreach (keys %$product) {
+            $settings->{uc $_} = $product->{$_} if $_ ne 'settings';
+        }
+        foreach my $s_key (keys %$product_settings) {
+            if ($s_key eq 'WORKER_CLASS') {
+                push @worker_class, $product_settings->{$s_key};
+                next;
+            }
+            $settings->{$s_key} = $product_settings->{$s_key};
+        }
+        if (my $machine = $job_template->{machine}) {
+            $settings->{MACHINE} = $machine;
+            if (my $mach = $machines->{$machine}) {
+                my $machine_settings = $mach->{settings} // {};
+                foreach (keys %$machine_settings) {
+                    if ($_ eq 'WORKER_CLASS') {
+                        push @worker_class, $machine_settings->{WORKER_CLASS};
+                        next;
+                    }
+                    $settings->{$_} = $machine_settings->{$_};
+                }
+                $settings->{BACKEND} = $mach->{backend} if $mach->{backend};
+                $settings->{PRIO} = $mach->{priority} // 50;
+            }
+        }
+        $settings->{WORKER_CLASS} = join ',', sort @worker_class if @worker_class > 0;
+        foreach (keys %$args) {
+            $settings->{uc $_} = $args->{$_} if $_ ne 'TEST';
+        }
+        $settings->{DISTRI} = lc($settings->{DISTRI}) if $settings->{DISTRI};
+
+        OpenQA::JobSettings::parse_url_settings($settings);
+        OpenQA::JobSettings::handle_plus_in_settings($settings);
+        my $error = OpenQA::JobSettings::expand_placeholders($settings);
+        $error_msg .= $error if defined $error;
```

Comment:
I think it is actually useful to present the user with a list of errors, e.g. like a compiler doesn't stop at the first error but shows you all the errors (in the current compilation unit). This way you don't have to invoke it over and over again and can instead tackle multiple problems in one go.

Reply:
Right, that's a fair point. I would still prefer to not schedule any jobs unless all of them are fine, though.

Reply:
> I would still prefer to not schedule any jobs unless all of them are fine, though.

That's actually how it works. In the code that is "gathering" different error messages no jobs are created visibly or even persistently yet. That happens only later and if errors occurred here we abort before it comes to this.

Reply:
Okay. I mentioned `schedule_iso` because it is documented to behave the way I describe it. If that's not what is happening here I'm happy to be corrected 😊

Reply:
> Returns a list of job ids from the jobs that were successfully scheduled                    
  and a list of failure reason for the jobs that could not be scheduled.

Specifically is what I was referring to

Reply:
Yes, but this distinction is not about problems that can happen *at this point*. So it is really out of scope of my changes. (And in case of problems like an unhandled exception one would also just get a 500 response instead of a half set of jobs being created as the actual creation of jobs in the DB happens in a transaction.)